x-pack/plugins/security/public/authentication/login/login_page.tsx

```diff
@@ -144,68 +175,120 @@ export class LoginPage extends Component<Props, State> {
-              />
-            }
-          />
-        );
-      default:
-        return (
-          <DisabledLoginForm
-            title={
-              <FormattedMessage
-                id="xpack.security.loginPage.unknownLayoutTitle"
-                defaultMessage="Unsupported login form layout."
-              />
-            }
-            message={
-              <FormattedMessage
-                id="xpack.security.loginPage.unknownLayoutMessage"
-                defaultMessage="Refer to the Kibana logs for more details and refresh to try again."
-              />
-            }
-          />
-        );
+    if (layout === 'error-es-unavailable') {
+      return (
+        <DisabledLoginForm
+          title={
+            <FormattedMessage
+              id="xpack.security.loginPage.esUnavailableTitle"
+              defaultMessage="Cannot connect to the Elasticsearch cluster"
+            />
+          }
+          message={
+            <FormattedMessage
+              id="xpack.security.loginPage.esUnavailableMessage"
+              defaultMessage="See the Kibana logs for details and try reloading the page."
+            />
+          }
+        />
+      );
+    }
+
+    if (layout === 'error-xpack-unavailable') {
+      return (
+        <DisabledLoginForm
+          title={
+            <FormattedMessage
+              id="xpack.security.loginPage.xpackUnavailableTitle"
+              defaultMessage="Cannot connect to the Elasticsearch cluster currently configured for Kibana."
+            />
+          }
+          message={
+            <FormattedMessage
+              id="xpack.security.loginPage.xpackUnavailableMessage"
+              defaultMessage="To use the full set of free features in this distribution of Kibana, please update Elasticsearch to the default distribution."
+            />
+          }
+        />
+      );
+    }
+
+    if (layout !== 'form') {
+      return (
+        <DisabledLoginForm
+          title={
+            <FormattedMessage
+              id="xpack.security.loginPage.unknownLayoutTitle"
+              defaultMessage="Unsupported login form layout."
+            />
+          }
+          message={
+            <FormattedMessage
+              id="xpack.security.loginPage.unknownLayoutMessage"
+              defaultMessage="Refer to the Kibana logs for more details and refresh to try again."
+            />
+          }
+        />
+      );
+    }
+
+    const loginSelector =
+      showLoginSelector &&
+      selector.providers.map((provider, index) => (
+        <EuiButton
+          key={index}
+          className="loginWelcome__selectorButton"
+          iconType="user"
+          fullWidth={true}
+          onClick={() => this.login(provider.type, provider.name)}
+        >
+          {provider.options.description ?? `${provider.type}/${provider.name}`}
```

Comment:
__note:__ that's what users are supposed to see when they use old providers config format + explicitly set `selector.enabled: true`.

Reply:
I like it, will change, thanks!

Reply:
Sorry, one revision. I think "Login" should actually be "Log in" to be consistent with the `BasicLoginForm`.

This makes me wonder: Should the SSO login buttons always start with "Log in with", even when a provider has a `description`? The buttons could read something like:

```ts
const description = provider.options.description ?? `${provider.type}/${provider.name}`;

<FormattedMessage 
  id="xpack.security.loginPage.loginProviderDescription" 
  defaultMessage="Log in with {description}" 
  values={{
    description
  }}
/>
```

Reply:
> Sorry, one revision. I think "Login" should actually be "Log in" to be consistent with the BasicLoginForm.

Will fix :+1: 

> This makes me wonder: Should the SSO login buttons always start with "Log in with", even when a provider has a description? The buttons could read something like:

Hmm don't have a strong opinion on that to be honest, was thinking about giving more freedom to admins since we support i18n only for 3 languages and `Log in with` may not play very well for all scenarios, but it's just a wild guess. If you feel strong about that - let me know, I'll change to always prefix with `Log in with..`.

Reply:
I don't have a strong opinion here, I was just wondering out loud